static/js/global.d.ts

```diff
@@ -7,4 +7,15 @@ declare let zulip_test: any;
 
 interface JQuery {
     expectOne(): JQuery;
+    tab(opts: string): void; // From static/third/bootstrap
+}
+
+interface HTMLElement {
+    // Non Standard IE 8-9 createdTextRange API used in ui_utils.ts
+    // because window.getSelection & document.createRange is not supported.
+    createTextRange?: () => {
+        collapse(action: boolean): void;
+        moveToElementText(el: HTMLElement): void;
+        select(): void;
+    };
```

Comment:
TypeScript already knows about `createTextRange` on `HTMLBodyElement`, `HTMLButtonElement`, `HTMLInputElement`, and `HTMLTextAreaElement`. You probably just need a type check somewhere.

Reply:
Oh I see, it was removed in microsoft/TypeScript@b0584b58fa07ee8c06e6d6f0f1bce2d4c37c7640. Anyway, we don’t support IE anymore, so we can remove this path.

Reply:
Merged #19014 :)

Reply:
Removed after rebase.